src/components/scenes/TradeOptionSelectScene.tsx

```diff
@@ -0,0 +1,414 @@
+  const icon = paymentTypeIcon ?? { uri: selectedQuote.partnerIcon }
+
+  // Determine custom title rendering
+  const customTitleKey = paymentTypeToCustomTitleKey[selectedQuote.paymentType]
+  const defaultTitle = getPaymentTypeDisplayName(selectedQuote.paymentType)
+
+  // Render custom title based on payment type
+  let titleComponent: React.ReactNode
+  switch (customTitleKey) {
+    case 'applepay':
+      // Per Apple branding guidelines, "Pay with" is NOT to be translated
+      titleComponent = (
+        <TitleAppleContainer>
+          <TitleText numberOfLines={1}>
+            {/* eslint-disable-next-line react-native/no-raw-text */}
+            {'Pay with '}
+          </TitleText>
+          <TitleAppleLogo source={paymentTypeLogoApplePay} />
+        </TitleAppleContainer>
+      )
+      break
+    default:
+      titleComponent = <TitleText numberOfLines={1}>{defaultTitle}</TitleText>
+  }
+
+  // Handle provider press - show modal to select between providers
+  const handleProviderPress = async () => {
+    if (quotes.length <= 1) {
+      // No other providers to choose from
+      return
+    }
+
+    // Create items array for the RadioListModal
+    const items = quotes.map(quote => {
+      // Format the crypto amount for each provider
+      // const localeAmount = formatNumber(toFixed(quote.cryptoAmount, 0, 6))
+      const amount =
+        quote.direction === 'buy' ? quote.fiatAmount : quote.cryptoAmount
+      const currencyCode =
+        quote.direction === 'buy'
+          ? quote.fiatCurrencyCode.replace('iso:', '')
+          : quote.displayCurrencyCode
+      const text = `(${amount} ${currencyCode})`
+
+      return {
+        name: quote.pluginDisplayName,
+        icon: quote.partnerIcon, // Already full path
+        text
+      }
+    })
+
+    const selectedName = await Airship.show<string | undefined>(bridge => (
+      <RadioListModal
+        bridge={bridge}
+        title={lstrings.trade_option_choose_provider}
+        items={items}
+        selected={selectedQuote.pluginDisplayName}
+      />
+    ))
+
+    if (selectedName != null) {
+      const selectedIndex = quotes.findIndex(
+        quote => quote.pluginDisplayName === selectedName
+      )
+      if (selectedIndex !== -1) {
+        setSelectedQuoteIndex(selectedIndex)
+      }
+    }
+  }
+
+  return (
+    <PaymentOptionCard
+      title={titleComponent}
+      icon={icon}
+      totalAmount={sprintf(
+        lstrings.string_total_amount_s,
+        `${selectedQuote.fiatAmount} ${fiatCurrencyCode} → ${selectedQuote.cryptoAmount} ${selectedQuote.displayCurrencyCode}`
+      )}
+      settlementTime={formatSettlementTime(selectedQuote.settlementRange)}
+      partner={{
+        displayName: selectedQuote.pluginDisplayName,
+        icon: { uri: selectedQuote.partnerIcon }
+      }}
+      renderRight={isBestRate ? () => <BestRateBadge /> : undefined}
+      onPress={async () => {
```

Comment:
[96b0de263] Avoid inline async in prop. Define const handlePress = useCallback(() => onPress(selectedQuote), [onPress, selectedQuote]) and pass onPress={handlePress}.